r/README.md

```diff
@@ -178,6 +178,28 @@ For any other build/configuration challenges, see the [C++ developer
 guide](https://arrow.apache.org/docs/developers/cpp/building.html) and
 `vignette("install", package = "arrow")`.
 
+### Using `remotes::install_github(...)` 
+
+If you need an Arrow installation from a specific repository or at a specific ref, `remotes::install_github()` should work on most platforms. This method is helpful if you need a full install of arrow that is separate from another install (e.g. we use this in [arrowbench](https://github.com/ursacomputing/arrowbench) to install development versions of arrow isolated from the system install). However there are some caveats to be aware of:
+
+* Setting the environment variable `FORCE_TOOLS_LIBS_SCRIPT` to `true` will avoid linking to any arrow libraries already installed and attempt to build arrow from the same source at the repository+ref given.
+* If you are using the `FORCE_TOOLS_LIBS_SCRIPT` you must also set `build = FALSE` in the `remotes::install_github()` call. This is similar to checking out the repository and calling `R CMD INSTALL .` in the `arrow/r` directory (as opposed to first calling `R CMD BUILD .` and then installing the tar.gz file that produces, which is the default for `remotes::install_github()`).
+* Specify `subdir = "r"` to get the R package.
+* On macOS you may need to also specify the environment variable `SDKROOT` to an appropriate location (typically something like `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk`). This is most easily and reliably done using `xcrun --show-sdk-path` (to set the environment variable inside of R you can `Sys.setenv(SDKROOT=system("xcrun --show-sdk-path", intern = TRUE))`).
```

Comment:
"may"? When/why?

Reply:
I've added more details here to explain what's going on. 

The biggest reason for "may" (beyond people on less than 10.14) is that I suspect many people have done things like symlink an sdk back to `/usr/include` which "avoids" this problem (but has its own issues) at the suggestion of various threads on SO about compiling issues. I've also seen many people suggest adding `export SDKROOT="$(xcrun --show-sdk-path)"` to one's profile to resolve issues like this.